interface.go

```diff
@@ -7,6 +7,10 @@ import (
 
 // Milter is an interface for milter callback handlers
 type Milter interface {
+	// Init is called on begin of a new Mail (before Connect() and after Body() )
+	// Can be used to Reset session state to before SMFIC_MAIL
+	Init()
```

Comment:
Idea was nice but realization is BAD. Let me describe.
You have two different context which live during session. First it's SMTP session context, which usually not changed, with Connection and HELO info. Second, is Email context which could changed from 0 to N times per one session. In your realization you force to use the same function in 3! different cases with 2 different context's:
1) When you actually start Session - that's looks fine, looks like it should be.
2) In the end of body... Let's see. How about most popular case when you have only one Email in session? After all job done you force to re-initialize object which normally be discarded. What for? 
Additionally you must be aware is that first call of `Init` or not?
3) On `Abort` command.  In doc says:  `SMFIC_ABORT` -   Resets internal state of milter program to before SMFIC_HELO, but keeps connection open. That's mean next command from client will be `SMFIC_HELO` instead of `SMFIC_MAIL` like in case above. What you will do if you need to make some job on abort command, how you will know is the current call was on abort action or not? 
It's too much for one function. I suggest handle only first case. Create a function `Abort` for third case. And left second case on library clients. Client should handle it by self, because library know nothing about client flow. Did his code have state or not... That's only client pain.

Reply:
Sorry i didnt updated my pull request here. I changed the behaviour in my latest master branch

https://github.com/mschneider82/milter

The changed were to much for a single PR

Reply:
Additionally you force implement methods `Init` `Disconnect` including cases when them not necessary. That's not idea of Milter when you could ignore not used method via `OptProtocol`
All initialized action you could make in `MilterInit` function. So, that's  not needed at all. 
And disconnect is better to switch on `Closer` interface

Reply:
The idea of Init() and not using MilterInit is to share the code, because it will be equal.
Feel free to suggest a API Change on my master fork. I also added Session/Mail ID's there.

I suggest to close this PR because after End Of Body was not a good idea